x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1117,6 +1165,58 @@ Collection<Object> createComponents(
         return components;
     }
 
+    private void applyPendingSecurityMigrations(
+        ClusterState clusterState,
+        FeatureService featureService,
+        SecurityIndexManager.State newState
+    ) {
+        if (clusterState.nodes().isLocalNodeElectedMaster() == false) {
+            return;
+        }
+
+        Integer nextMigrationVersion = SecurityMigrations.MIGRATIONS_BY_VERSION.higherKey(newState.migrationsVersion);
+
+        // Check if next migration that has not been applied is eligible to run on the current cluster
+        boolean eligibleSecurityMigrationExists = nextMigrationVersion != null
+            && isEligibleSecurityMigration(
+                loadIndexMappingVersion(SECURITY_MAIN_ALIAS, clusterState),
```

Comment:
Can we please expose the current index mapping version from `SecurityIndexManager.State newState` (I know there are 2 mapping versions there, a clean-up would be very welcomed!).

Reply:
Good idea!

Reply:
Don't forget to set this to `newState.indexMappingVersion`.